saw-central/src/SAWCentral/Crucible/MIR/ResolveSetupValue.hs

```diff
@@ -255,6 +247,23 @@ buildMirAggregateWithVal sym elems vals f =
         Nothing -> fail $ "buildMirAggregateWithVal: type mismatch at offset "
           ++ show off ++ ": expected " ++ show shp ++ ", but the MIRVal contained " ++ show shp'
 
+buildMirAggregateArrayWithVal ::
+  (Monad m, MonadFail m) =>
+  Sym ->
+  Word ->
+  TypeShape tp ->
+  Word ->
+  [MIRVal] ->
+  (Word -> RegValue Sym tp -> m (RegValue Sym tp)) ->
```

Comment:
It would be worth adding some Haddock comments to distinguish all of these different `Word`s in the type signature.

Reply:
Added doc comments in 22cbe9ce098d677eaf4045013fab05bc7f9614cc